src/CounterWithImmutables.sol

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+contract CounterWithImmutables {
+  uint256 public immutable scaleFactor;
```

Comment:
nit: technically by our normal conventions this should SCALE_FACTOR right?

Reply:
Updated in 0ddde20